tests/unit/remote/test_http.py

```diff
@@ -13,3 +16,14 @@ def test_no_traverse_compatibility(dvc_repo):
 
     with pytest.raises(ConfigError):
         RemoteHTTP(dvc_repo, config)
+
+
+def test_download_fails_on_error_code(dvc_repo):
+    with StaticFileServer() as httpd:
+        url = "http://localhost:{}/".format(httpd.server_port)
+        config = {"url": url}
+
+        remote = RemoteHTTP(dvc_repo, config)
+
+        with pytest.raises(HTTPError):
+            remote._download(URLInfo(url) / "file.txt", "file.txt")
```

Comment:
Maybe use `"missing.txt"` to make it more obvious.

Reply:
done